tests/phpunit/tests/comment/getCommentAuthorLink.php

```diff
@@ -0,0 +1,81 @@
+<?php
```

Comment:
I plucked this file from another PR that I got side tracked with when I started working on this. I also plucked a bit of code from that PR on the function `get_comment_author_link()` to prevent the notices thrown.

See trac https://core.trac.wordpress.org/ticket/57615 and #3980

I realize this now has conflicting code in two semi-related PRs, I can pluck the code out of the other PR or merge it all together but I wanted to keep this PR smaller. But... I also had to add a new tests file to test the changes in the function in this PR... well you know